cdci_data_analysis/analysis/parameters.py

```diff
@@ -341,241 +350,203 @@ def reprJSON(self):
-                                  check_value=self.check_name_value,
-                                  name=name,
-                                  allowed_units=_allowed_units)
+        super().__init__(value=value,
+                         units=name_format,
+                         check_value=self.check_name_value,
+                         name=name,
+                         allowed_units=_allowed_units)
 
     @staticmethod
     def check_name_value(value, units=None, name=None):
         pass
 
 
 class Float(Parameter):
-    def __init__(self,value=None,units=None,name=None):
-
-        _allowed_units = None
-
-        #wtform_dict = {'keV': FloatField}
-
-        super(Float, self).__init__(value=value,
-                                   units=units,
-                                   check_value=self.check_float_value,
-                                   name=name,
-                                   allowed_units=_allowed_units)
-                                   #wtform_dict=wtform_dict)
-
-        self.value=value
+    def __init__(self, value=None, units=None, name=None, allowed_units=None, default_units=None, check_value=None):
+
+        # wtform_dict = {'keV': FloatField}
+        if check_value is None:
+            check_value = self.check_float_value
+
+        super().__init__(value=value,
+                         units=units,
+                         check_value=check_value,
+                         default_units=default_units,
+                         name=name,
+                         default_type=float,
+                         allowed_units=allowed_units)
+        # wtform_dict=wtform_dict)
 
     @property
     def value(self):
         return self._v
 
     @value.setter
     def value(self, v):
-        if v is not None and v!='':
+        if v is not None and v != '':
             self.check_float_value(v, name=self.name, units=self.units)
-            self._v = np.float(v)
-
+            self._v = float(v)
         else:
-            self._v=None
+            self._v = None
+
+    def get_value_in_default_format(self):
+        return float(self.value)
 
     @staticmethod
     def check_float_value(value, units=None, name=None):
-        #print('check type of ',name,'value', value, 'type',type(value))
-        if value is None or value=='':
+        # print('check type of ',name,'value', value, 'type',type(value))
+        if value is None or value == '':
             pass
         else:
             try:
-              value=ast.literal_eval(value)
+                float(value)
             except:
-                pass
-            value=np.float(value)
-            if type(value) == int or type(value) == np.int:
-                pass
-            elif type(value) == float or type(value) == np.float:
-                pass
-            else:
-                raise RuntimeError('type of ', name, 'not valid', type(value))
+                raise RuntimeError(f'type {type(value)} not valid for {name}')
```

Comment:
If `value = 'bla'` the message will say `str not valid`, but `str` can be valid if `value = '1.1'`. It's not just about the  type, value also matters.
So could we make this longer, to help debug, adding the actual value in the message along with its type?

Reply:
yeah it makes sense, didnt think about it, also for the Integer class, have to consider the same case